contentprovider.go

```diff
@@ -50,13 +54,17 @@ func (p *contentProvider) setDocument(docID uint32) {
 	p._nl = nil
 	p._sects = nil
 	p._data = nil
+	p.docSectionsDone = false
 }
 
 func (p *contentProvider) docSections() []DocumentSection {
-	if p._sects == nil {
+	// It is not sufficient to test for p._sects == nil, because for documents
+	// without symbols p._sects = nil, and we always ignore the cache.
+	if !p.docSectionsDone {
```

Comment:
alternative implementation to avoid tracking extra state:

if there are no symbols in the document, won't readDocSections return something empty? ie if `d.docSectionsIndex[i] ==  d.docSectionsIndex[i+1]` you can early return? I haven't dived deep enough into this idea to know if it is feasible

Also is this done to fix a regression?

Reply:
Because I changed the way we call findSection, some of tests reported changed counts for contentBytesLoaded. The root casue was that for empty symbol we count 1 byte and don't fill the cache. In other words, this fixes a minor issue with the cache which showed up during testing.

Reply:
Can't you just do a check for empty here and then avoid loading anything at all? IE your contentBytesLoaded will go down.